src/app/products/share-products/dividends-share-product/dividends.components.ts

```diff
@@ -0,0 +1,31 @@
+import { Component, OnInit, ViewChild } from '@angular/core';
+import { MatPaginator, MatSort, MatTableDataSource } from '@angular/material';
+import { ActivatedRoute } from '@angular/router';
+
+@Component({
+  selector: 'mifosx-dividends-share-product',
+  templateUrl: './dividends.component.html',
+  styleUrls: ['./dividends.component.scss']
+})
+export class DividendsComponent implements OnInit {
+  dividendsData: any;
+  displayedColumns: string[] = ['name', 'startDividend', 'endDividend', 'status'];
+  dataSource: MatTableDataSource<any>;
+
+  @ViewChild(MatPaginator) paginator: MatPaginator;
+  @ViewChild(MatSort) sort: MatSort;
+
+  constructor() { }
```

Comment:
You haven't connected the component to the API?, you have to construct API call in the service, use a resolver for it and then subscribe to data from route, see other simple table implementations in the app.
For which api endpoint to use you can open network tab in browser navigate to the same page in demo.mifos.io or staging.mios.io and see which API request got triggered upon navigation.